insights/parsers/vdsm_log.py

```diff
@@ -29,66 +28,108 @@ class VDSMLog(LogFileOutput):
 
         Yields:
-            Dictionaries corresponding to the data in each line split into
-            functional elements:
-
-            * **raw_line** - the original unparsed line
-            * **thread** - the thread or task ID
-            * **level** - the log level
-            * **timestamp** - the date of the log line (as a string)
-            * **datetime** - the date as a datetime object (if conversion is possible)
-            * **module** - the module logging the message
-            * **line** - the line reporting this message within the module
-            * **logname** - the logger's name
-            * **funcname** - the function name reporting the message
-            * **message** - the body of the message
-
-            The **datetime** field is only present if the `timestamp` field
-            is present contains a time stamp in the valid format.
-
-            If the line is too short, for some reason, then as many fields as
-            possible are pulled from the line.
+            Dictionary with following keys
+
+            * **asctime(datetime)** - date and time as datetime object
+            * **level(str)** - log level. Can me INFO, ERROR, WARN or DEBUG
+            * **thread(str)** - thread name
+            * **logname(str)** - filter name
+            * **message(str)** - the body of the message
+            * **module(str)** - module name
+            * **lineno(str)** - line number which triggered the log
+
+        This will NOT parse Python Traceback. Any unparsed line(s) will be yield as a list
         """
+        time_format = '%Y-%m-%d %H:%M:%S,%f'
+
+        re_datetime_obj = re.compile(r"^(\d{4}-\d{2}-\d{2}\s+[\d{2}:]+\d{2},\d{3})")  # noqa
+        re_name_obj = re.compile(r"\[(\w.+)\]")
+        re_thread_and_module = re.compile(r"\((.*?)\)")
+        re_message_obj = re.compile(r"\[\w.+\](.*?)\(\w+\:\w+\)$")
+        re_level_obj = re.compile(r"INFO|ERROR|DEBUG|WARN")
 
-        # funcname is in brackets before message, so parse those separately
-        fieldnames = ('thread', 'level', 'timestamp', 'module', 'line', 'logname')
         for line in lines:
-            parts = line.split('::', 6)
-            fields = {'raw_line': line}
-            fields.update({k: v for k, v in zip(fieldnames, parts)})
-            if len(parts) == 7:
-                func, msg = parts[6].split(' ', 1)
-                fields['funcname'] = func[1:-1]  # remove surrounding brackets
-                fields['message'] = msg
-            # Did we get a timestamp in there?
-            if 'timestamp' in fields:
-                # Try to convert the datetime if possible
-                try:
-                    fields['datetime'] = datetime.strptime(
-                        fields['timestamp'][:-4],  # skip milliseconds
-                        self.time_format
-                    )
-                except:
-                    pass
-            yield fields
+            if re_datetime_obj.match(line):
```

Comment:
Per our discussion on using the regex, please double confirm the input data when doing the parsing job. And please add more description on the regex rule.